### ISSUE TITLE
Program crashes after connect and initial draw

### ISSUE THREAD

**Issue by spotlesscoder** [VolosR/TTGOWeatherStation#13]
First of all, thank you for the amazing idea and putting it online

I tried running this on two different Lilygo T Display S3 but unfortunately it crashes after a few milliseconds and the microcontroller reboots.

The picture and placeholders are sometimes correctly displayed for the fraction of a second but no live data and then the microcontroller reboots, goes to the WiFi connection screen again.

This happens in an endless loop :/

**Comment by JuniorPolegato** [comment]
Hi!

I forked and refactoring this project, [here](https://github.com/JuniorPolegato/TTGOWeatherStation)

Can you try this one?

[]'s

Junior Polegato

**Comment by orion14794** [comment]
> Hi!
> 
> I forked and refactoring this project, [here](https://github.com/JuniorPolegato/TTGOWeatherStation)
> 
> Can you try this one?
> 
> []'s
> 
> Junior Polegato

I am getting same issues. Did exactly as you instructed, device enters an infinite loop showing only connecting in white words and a black background, then reboots doing this loops over and over. Reboots every 3 seconds. Never showing the graphics/interface like it's shown in your posted photo...

**Comment by JuniorPolegato** [comment]
Hi!

Maybe a S3 issue, did you select the correct board?
You can try to enable debugging verbose and see messages on the serial monitor.

**Comment by orion14794** [comment]
> Hi!
> 
> Maybe a S3 issue, did you select the correct board? You can try to enable debugging verbose and see messages on the serial monitor.

Yes. I've selected proper board and all libraries needed. It's not the 1st sketch I've flashed on my LiliGo T-Display ESP 32 S3, touch version. I've flashed Volos sketches before example: The PCB board  clock display sketch which is currently running on my device...

**Comment by JuniorPolegato** [comment]
> > Hi!
> > Maybe a S3 issue, did you select the correct board? You can try to enable debugging verbose and see messages on the serial monitor.
> 
> Yes. I've selected proper board and all libraries needed. It's not the 1st sketch I've flashed on my LiliGo T-Display ESP 32 S3, touch version. I've flashed Volos sketches before example: The PCB board clock display sketch which is currently running on my device...

Hi!

I do many tests and got this problem when occur connection errors, invalid key (personal key to API from OpenWeather) or invalid city/country (must be like at OpenWeather).

So I solve this with a loop around this errors in new version of my fork here: [TTGOWeatherStation](https://github.com/JuniorPolegato/TTGOWeatherStation).

In this version, you can manage WiFi, key and Cities from web browser. I am using [ESPUserConnection](https://github.com/JuniorPolegato/ESPUserConnection) of my own.
